### PR TITLE
Check and fix broken anchor links.

### PR DESCRIPTION
**Pull request by dennisreimann**
In addition to #1682, this also checks URL fragments so that we also get to know about broken anchors.

### CODE REVIEW

File: BTCPayServer.Tests/UnitTest1.cs [hunk 1]
```diff
@@ -155,16 +155,33 @@ private static async Task CheckLinks(Regex regex, HttpClient httpClient, string
 
         private static async Task AssertLinkNotDead(HttpClient httpClient, string url, string file)
         {
+            var uri = new Uri(url);
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:158]: Could be a malformed url that causes the Uri constructor to crash, so declare it out here and set inside the try block

Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:158]:
```suggestion
            Uri uri = null;
```

File: BTCPayServer.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -155,16 +155,33 @@ private static async Task CheckLinks(Regex regex, HttpClient httpClient, string
...
+
             try
             {
-                using var request = new HttpRequestMessage(HttpMethod.Get, new Uri(url));
+                using var request = new HttpRequestMessage(HttpMethod.Get, uri);
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:162]:
```suggestion
                uri = new Uri(url);
                using var request = new HttpRequestMessage(HttpMethod.Get, uri);
```

File: BTCPayServer.Tests/UnitTest1.cs [hunk 3]
```diff
@@ -155,16 +155,33 @@ private static async Task CheckLinks(Regex regex, HttpClient httpClient, string
...
+                    var contents = await response.Content.ReadAsStringAsync();
+                    Assert.Matches($"id=\"{fragment}\"", contents);
+                }
+
                 Logs.Tester.LogInformation($"OK: {url} ({file})");
             }
+            catch (Exception ex) when (ex is MatchesException)
+            {
+                var details = ex.Message;
+                Logs.Tester.LogInformation($"FAILED: {url} ({file}) – anchor not found: {uri.Fragment}");
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:181]:
```suggestion
                Logs.Tester.LogInformation($"FAILED: {url} ({file}) – anchor not found: {(uri?.Fragment??string.Empty)}");
```

File: BTCPayServer.Tests/UnitTest1.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer.Tests/UnitTest1.cs:158]: That's fine if it crash, because it SHOULD crash.

**Comment by NicolasDorier** [comment]
Merging this, I think that the point of crash @Kukks  mentioned are not problematic, as it will crash with proper error message.

**Pull request merged**